### PR TITLE
Feat(eth-wallet-contract): reproducible build

### PR DESCRIPTION
**Pull request by birchmd**
Continuing with the work on eth-wallet contract, this PR adds a reproducible build process. It uses the Docker image [recommended in near-sdk-rs](https://github.com/near/near-sdk-rs#building-with-reproducible-builds). Note that the reproducible build process is skipped if the wasm file is already present and it is still committed to the repo. This is intentional because it means developers not wanting to worry about the eth-wallets contract can safely ignore it entirely when setting up their development environment. However, it does mean that any changes to the wallet contract implementation will also require locally deleting the wasm file and then running the build process to generate the new wasm file. The test that checks the hash of the wasm file will also need to be updated with whatever the new hash is.

For reviewers: please try deleting the wasm file yourself and running the build process (e.g. with `cargo test -p near-wallet-contract --features nightly`) to ensure that it is indeed reproducible across machines.

This PR also sets the value of `ADDRESS_REGISTRAR_ACCOUNT_ID` equal to the agreed upon value `address-map.near`. This account exists on both testnet and mainnet so the same value can be used for both networks. For now Aurora holds the keys to this account, but once we are sure the address registrar contract implementation (present in the previous eth wallet PR) will not change (e.g. after it has been audited) then we will delete the keys from that account to lock the contract implementation.

### CODE REVIEW

File: runtime/near-wallet-contract/build.rs [hunk 1]
```diff
@@ -41,23 +41,25 @@ fn build_contract(dir: &str, args: &[&str], output: &str) -> Result<()> {
 /// process, to avoid unexpected behaviors due to the workspace configurations.
 // TODO(eth-implicit) Change it to have a reproducible hash of the WASM file.
 // see https://github.com/near/nearcore/pull/10269#discussion_r1430139987.
```
Comment (staffik) [inline runtime/near-wallet-contract/build.rs:43]: Can now remove this comment.

**Review by staffik — APPROVED** [review]
LGTM
Can we add instruction `cargo test -p near-wallet-contract --features nightly` to README?
I am asking because `cargo build --features nightly` did not work because of `include_bytes` in `read_contract`.

**Comment by birchmd** [comment]
> `cargo build --features nightly` did not work

@staffik could you elaborate more on the error you got? This command does work on my machine. It works whether or not `runtime/near-wallet-contract/res/wallet_contract.wasm` is present (when it is not present it is re-built from the logic in `runtime/near-wallet-contract/build.rs`)

File: runtime/near-wallet-contract/build.rs (hunk 1 above)
Comment (birchmd) [reply runtime/near-wallet-contract/build.rs]: Done in https://github.com/near/nearcore/pull/11011/commits/b279f67d496b449349e54042abfe13e9d383000f

**Comment by staffik** [comment]
> > `cargo build --features nightly` did not work
> 
> @staffik could you elaborate more on the error you got? This command does work on my machine. It works whether or not `runtime/near-wallet-contract/res/wallet_contract.wasm` is present (when it is not present it is re-built from the logic in `runtime/near-wallet-contract/build.rs`)

I've removed `runtime/near-wallet-contract/res/wallet_contract.wasm` and run `cargo build --features nightly` on MacOS.
This is my output:
```
% cargo build --features nightly
   Compiling near-store v0.0.0 (/Users/stafik/near/nearcore/core/store)
   Compiling near-client-primitives v0.0.0 (/Users/stafik/near/nearcore/chain/client-primitives)
   Compiling near-wallet-contract v0.0.0 (/Users/stafik/near/nearcore/runtime/near-wallet-contract)
   Compiling neard v0.0.0 (/Users/stafik/near/nearcore/neard)
   Compiling congestion-model v0.0.0 (/Users/stafik/near/nearcore/tools/congestion-model)
error: couldn't read runtime/near-wallet-contract/src/../res/wallet_contract.wasm: No such file or directory (os error 2)
  --> runtime/near-wallet-contract/src/lib.rs:14:16
   |
14 |     let code = include_bytes!("../res/wallet_contract.wasm");
   |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `include_bytes` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `near-wallet-contract` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

**Comment by birchmd** [comment]
Hmm, that is strange. I do not know why the build logic would not trigger to create the missing file. I know it's time-consuming to do this, but when it's not an interruption for you to do so (like when you're about to go for a break or something), could you try running `cargo clean` and then `cargo build --features nightly`? Maybe the problem is somehow on MacOS it is not properly detecting that the missing file should trigger the build process to happen.

**Comment by staffik** [comment]
> runtime/near-wallet-contract/res/wallet_contract.wasm

The same issue even when I did `cargo clean`.
I had similar issue on CI, see https://github.com/near/nearcore/pull/10269#discussion_r1428375728.

**Comment by birchmd** [comment]
Weird that even `cargo clean` does not help. I kind of expect the current solution not to work in CI because it uses a docker image and I think setting up the CI to run the docker image would take additional work.

I'm doing a bit of refactoring of the build process in the next PR I have planned. So I think I'll merge this PR for now and we can continue debugging this in the next PR.

**Pull request merged**